library of setup and function definitions

debobiswas01 opened the pull request:
Library containing utilities_setup and utilities_functions

function names all have lower case now
__init__.py file imports all functions
scan_field function defined separately for with and without keithley
scan_field_with_keithley has new input parameters, default values of input parameters set to values we were using previously.

debobiswas01 committed:
library of setup and function definitions 

Library containing utilities_setup and utilities_functions

comp-phys-marc commented on utilities/utilities_setup.py:
```diff
@@ -0,0 +1,14 @@
+import MultiPyVu as mpv
+from enum import Enum, auto
+
+class MVUInstrumentList(Enum):
```
Let's please put this setup in a function so we can call it from our scripts.

debobiswas01 replied:
changed it to a function. Note the file name is also now changed to setup.py

comp-phys-marc commented on utilities/utilities_functions.py:
```diff
@@ -0,0 +1,129 @@
+def save__temp_field_chamber_res_data():
```
Maybe this function belongs in a separate file because it is not so like the other two?

debobiswas01 replied:
yes that is a good point. I changed it to another file names save_data_functions.py and created a pull request for that

comp-phys-marc replied:
Hi Debo. Thanks for doing this. However, we would normally not create a pull request for a change that is requested in a PR that already exists (like this one). Instead, you just commit the change to *this* branch. :)

debobiswas01 replied:
thanks for that feedback. I have now committed the change to this branch and also removed the other PR that I had created. 
added this function to a save_data_functions.py file

comp-phys-marc commented on utilities/__init__.py:
```diff
@@ -0,0 +1,4 @@
+import utilities_setup
+from utilities_functions import save__temp_field_chamber_res_data
+from utilities_functions import scan_field_no_keithley
```
You can combine this line and the next line into one.

debobiswas01 replied:
```suggestion

from scan_functions import scan_field_no_keithley, scan_field_with_keithley
```

comp-phys-marc commented on utilities/__init__.py:
```diff
@@ -0,0 +1,4 @@
+import utilities_setup
```
It is a bit redundant to name the files in the utilities folder `utilities_<name>`. Can we just go with `<name>`? What do you think?

debobiswas01 replied:
good suggestion. I changed the file name

comp-phys-marc reviewed (changes requested):
Just a few changes requested.

debobiswas01 committed:
renamed utilities_functions.py to functions.py

debobiswas01 committed:
renamed utilities_setup.py to setup.py

debobiswas01 committed:
renamed functions.py to scan_functions.py

debobiswas01 committed:
put the setup in a function

put the setup commands in a function called instrument_ppmsvu_setup()

debobiswas01 committed:
response to feedback

created new file and moved the save_temp_field_chamber_res_data() function to this

debobiswas01 committed:
removed save_data type functions

response to feedback, removed the save_data type functions to a new file

debobiswas01 committed:
updated changed file names in __init__.py

based on response to feedback on changing file names, updated the changes on __init__.py as well

comp-phys-marc reviewed (approved):
Good work, addressed all feedback.
